### PR TITLE
[wasm][debugger] GetPossibleBreakpoints - return error message, as response

### PR DESCRIPTION
**Pull request by radical**
Also, adds some tests, though it isn't meant to be exhaustive tests for `GetPossibleBreakpoints`

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/ReturnAsErrorException.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public Result Error { get; }
+
+        public ReturnAsErrorException(JObject error)
+            => Error = Result.Err(error);
+
+        public ReturnAsErrorException(string message)
+            => Error = Result.Err(message);
+
+        public static ReturnAsErrorException ErrorObject(string message, string className)
```
Comment (lewing) [inline src/mono/wasm/debugger/BrowserDebugProxy/ReturnAsErrorException.cs:19]: should we make classname an enum and just make this another constreuctor?

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs [hunk 2]
```diff
@@ -878,7 +878,7 @@ public List<SourceLocation> FindPossibleBreakpoints(SourceLocation start, Source
             if (start.Id != end.Id)
             {
                 logger.LogDebug($"FindPossibleBreakpoints: documents differ (start: {start.Id}) (end {end.Id}");
-                return null;
+                throw new ReturnAsErrorException($"FindPossibleBreakpoints: documents differ (start: {start.Id}) (end {end.Id}");
```
Comment (lewing) [inline src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:881]: Is this the same error js throws?

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs [hunk 3]
```diff
@@ -239,14 +239,23 @@ protected override async Task<bool> AcceptCommand(MessageId id, string method, J
...
-                        var end = SourceLocation.Parse(args?["end"] as JObject);
-                        if (start != null && end != null && await GetPossibleBreakpoints(id, start, end, token))
-                            return true;
+                        try
+                        {
+                            var start = SourceLocation.Parse(args?["start"] as JObject);
+                            //FIXME support variant where restrictToFunction=true and end is omitted
+                            var end = SourceLocation.Parse(args?["end"] as JObject);
+                            if (start == null || end == null)
+                                throw new ReturnAsErrorException("Both start, and end need to be specified");
```
Comment (lewing) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:248]: https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-getPossibleBreakpoints end is actually optional.  I don't like returning custom errors on limitations that are ours

(see also https://github.com/mono/mono/pull/19999)

**Review by lewing — COMMENTED** [review]
This looks pretty good, have you tested to make sure the frontends handle the slightly different errors here?

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs (hunk 2 above)
Comment (radical) [reply src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:881]: Probably not. I just surfaced the errors that we already had. I can check what js returns for that, and do the same thing.

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs (hunk 3 above)
Comment (radical) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:248]: Ah, forgot about this one. I can integrate that here.

**Pull request closed without merging**